packages/app-desktop/gui/Sidebar/listItemComponents/ExpandIcon.tsx

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import { _ } from '@joplin/lib/locale';
+
+type ExpandIconProps = {
+	isExpanded: boolean;
+	isVisible: true;
+	targetTitle: string;
+}|{
+	isExpanded: boolean;
+	isVisible: false;
+	targetTitle?: string;
+};
```

Comment:
```suggestion
type ExpandIconProps = {
	isExpanded: boolean;
	isVisible: boolean;
	// For accessibility
	targetTitle?: string;
};
```

I'm not sure I like using a union type here. It was intended to make `targetTitle` only required when `isVisible` is `true`, but I think making it optional could be less confusing to someone reading the code.

Reply:
this can also be handled like this:
```ts
type VisibleIconProps = {
	isExpanded: boolean;
	isVisible: true;
	targetTitle: string;
};
type InvisibleIconProps = {
	isExpanded: boolean;
	isVisible: false;
	targetTitle?: string;
};
type ExpandedIconProps = VisibleIconProps | InvisibleIconProps;
```
But ultimately it comes down to complexity preference.

Reply:
> I'm not sure I like using a union type here. It was intended to make `targetTitle` only required when `isVisible` is `true`, but I think making it optional could be less confusing to someone reading the code.

I looked at this code for a little while too when I reviewed the code and I felt it was ok as it was. I guess the interface is small enough for this but for a bigger interface we'd probably want to avoid the duplication even if it means it's less correct.